src/d3a/d3a_core/redis_connections/aggregator_connection.py

```diff
@@ -68,6 +67,7 @@ def _delete_not_owned_devices(self, indict: dict, aggregator_uuid: str, outdict:
     def _create_grid_tree_event_dict(self, aggregator_uuid: str) -> dict:
         """Accumulate area_stats_tree_dict information and initiate a event dictionary
         to be sent to the client"""
+        from d3a.d3a_core.singletons import external_global_statistics
```

Comment:
I assume the lazy import is required in order to be able to instantiate the Redis connection only if needed, right ?

Reply:
Not really, there is here to solve a  circular import issue
I had the map of the imports on paper, I can draw you the whole setup if needed

Reply:
Nit: do you think there are other ways to solve this? Circular imports often point out some issues with the package structure so it can be good to deal with it. Not blocking though.

Reply:
Yes sure I prefer restructuring the modules, maybe in the context of another ticket

Reply:
I also agree with Pierpaolo, that in the long run we should restructure the modules to avoid the circular import, not a blocker for now though. In general, we have the implicit rule to only allow lazy imports in the d3a-web integration tests (e.g. in order to avoid imports from models.py before the test suite started Django), we are not too strict on this rule though therefore it is okay.